src/deployCommand.ts

```diff
@@ -47,18 +54,39 @@ export abstract class DeployCommand extends SourceCommand {
   protected setStash(deployId: string): void {
     const file = this.getStash();
     this.logger.debug(`Stashing deploy ID: ${deployId} in ${file.getPath()}`);
-    file.writeSync({ [DeployCommand.STASH_KEY]: { jobid: deployId } });
+    file.writeSync({
+      [this.getStashKey()]: { jobid: deployId },
+    });
+  }
+
+  /**
+   * This method is here to provide a workaround to stubbing a constructor in the tests.
+   *
+   * @param id
+   */
+  protected createDeploy(id?: string): MetadataApiDeploy {
+    return new MetadataApiDeploy({ usernameOrConnection: this.org.getUsername(), id });
+  }
+
+  protected getStashKey(): string {
+    return this.isSourceStash ? DeployCommand.SOURCE_STASH_KEY : DeployCommand.MDAPI_STASH_KEY;
   }
 
-  protected resolveDeployId(id: string): string {
+  protected resolveDeployId(id?: string): string {
     let stash: ConfigFile<StashFile>;
     if (id) {
       return id;
     } else {
       try {
         stash = this.getStash();
         stash.readSync(true);
-        const deployId = asString((stash.get(DeployCommand.STASH_KEY) as { jobid: string }).jobid);
+        const deployId = asString(
```

Comment:
why does this need `asString`?  there's already a type assertion inside it.

[I'd prefer to avoid these non-standard TS "helpers" wherever we can to keep things more approachable for externals]